tests/capture/transforms/test_capture_graph_decompose.py

```diff
@@ -0,0 +1,227 @@
+        assert interpreter.gate_set_contains(qml.RX(1.5, 0))
+        assert interpreter.gate_set_contains(qml.RZ(1.5, 0))
+        assert interpreter.gate_set_contains(qml.CNOT(wires=[0, 1]))
+        assert not interpreter.gate_set_contains(qml.Hadamard(0))
+
+    @pytest.mark.unit
+    def test_callable_gate_set_not_supported(self):
+        """Tests that specifying the gate_set as a function raises an error."""
+
+        with pytest.raises(TypeError, match="Specifying gate_set as a function"):
+            DecomposeInterpreter(gate_set=lambda op: op.name in {"RX", "RZ", "CNOT"})
+
+    @pytest.mark.integration
+    def test_fall_back(self):
+        """Tests that op.decompose() is used for ops unsolved in the graph."""
+
+        class CustomOp(qml.operation.Operation):  # pylint: disable=too-few-public-methods
+            """Dummy custom op."""
+
+            resource_keys = set()
+
+            @property
+            def resource_params(self):
+                return {}
+
+            def decomposition(self):
+                return [qml.H(self.wires[1]), qml.CNOT(self.wires), qml.H(self.wires[1])]
+
+        @qml.register_resources({qml.CZ: 1})
+        def my_decomp(wires, **__):
+            qml.CZ(wires=wires)
+
+        @DecomposeInterpreter(gate_set={"CNOT", "Hadamard"}, fixed_decomps={CustomOp: my_decomp})
+        def f():
+            CustomOp(wires=[0, 1])
+
+        jaxpr = jax.make_jaxpr(f)()
+        collector = CollectOpsandMeas()
+        collector.eval(jaxpr.jaxpr, jaxpr.consts)
+        assert collector.state["ops"] == [qml.H(1), qml.CNOT(wires=[0, 1]), qml.H(1)]
+
+    @pytest.mark.integration
+    def test_gate_set_targeted_decompositions(self):
+        """Tests that a simple circuit is correctly decomposed into different gate sets."""
+
+        def f(x, y, z):
+            qml.H(0)
+            qml.Rot(x, y, z, wires=0)
+            qml.MultiRZ(x, wires=[0, 1, 2])
+
+        decomposed_f = DecomposeInterpreter(gate_set={"Hadamard", "CNOT", "RZ", "RY"})(f)
+        jaxpr = jax.make_jaxpr(decomposed_f)(0.1, 0.2, 0.3)
+        collector = CollectOpsandMeas()
+        collector.eval(jaxpr.jaxpr, jaxpr.consts, 0.1, 0.2, 0.3)
+        assert collector.state["ops"] == [
+            # H is in the target gate set
+            qml.H(0),
+            # Rot decomposes to ZYZ
+            qml.RZ(0.1, wires=[0]),
+            qml.RY(0.2, wires=[0]),
+            qml.RZ(0.3, wires=[0]),
+            # Decomposition of MultiRZ
+            qml.CNOT(wires=[2, 1]),
+            qml.CNOT(wires=[1, 0]),
+            qml.RZ(0.1, wires=[0]),
+            qml.CNOT(wires=[1, 0]),
+            qml.CNOT(wires=[2, 1]),
+        ]
+
+        decomposed_f = DecomposeInterpreter(gate_set={"RY", "RZ", "CZ", "GlobalPhase"})(f)
+        jaxpr = jax.make_jaxpr(decomposed_f)(0.1, 0.2, 0.3)
+        collector = CollectOpsandMeas()
+        collector.eval(jaxpr.jaxpr, jaxpr.consts, 0.1, 0.2, 0.3)
+        assert collector.state["ops"] == [
```

Comment:
Is this long true list required for the purpose of this test? If so mind having some log in notion for sake of future maintenance?

Reply:
Yeah, I want to verify here that the circuit is decomposed to the correct list of ops.